src/db.c

```diff
@@ -1156,9 +1156,20 @@ void copyCommand(client *c) {
         case OBJ_ZSET: newobj = zsetDup(o); break;
         case OBJ_HASH: newobj = hashTypeDup(o); break;
         case OBJ_STREAM: newobj = streamDup(o); break;
-        case OBJ_MODULE:
-            addReplyError(c, "Copying module type object is not supported");
-            return;
+        case OBJ_MODULE: {
+                moduleValue *mv = o->ptr;
```

Comment:
Sort of a nitpick, but we don't encapsulate module logic that well. It seems like this should be closer to:
```
case OBJ_MODULE: 
    newobj = moduleDupOrReply(c, o);
    if (!newobj) return;
    break;
```

Reply:
I'm actually happy with this kind of nitpicking, it's not consistent with how it's done else where but heck, LIBTYFI.

Reply:
I agree that's how it's done elsewhere, but I think that is probably an area worth refactoring too :)